storm-client/src/jvm/org/apache/storm/security/auth/workertoken/WorkerTokenAuthorizer.java

```diff
@@ -95,9 +95,12 @@ private static IStormClusterState buildStateIfNeeded(Map<String, Object> conf, T
             throw new IllegalArgumentException("Token is not valid, token has expired.");
         }
 
-        PrivateWorkerKey key = keyCache.getUnchecked(deser);
-        if (key == null) {
-            throw new IllegalArgumentException("Token is not valid, private key not found.");
+        PrivateWorkerKey key;
+        try {
+            key = keyCache.getUnchecked(deser);
+        } catch (CacheLoader.InvalidCacheLoadException e) {
+            //This happens when the cache has a null returned to it.
```

Comment:
not sure about this. If `getUnchecked(deser);` returns null, it will throw an exception?

Reply:
I'll update the comment.